httpclient-processor/src/main/java/org/springframework/cloud/stream/module/httpclient/HttpClientProcessorConfiguration.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.httpclient;
+
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.config.SpelExpressionConverterConfiguration;
+import org.springframework.cloud.stream.messaging.Processor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * @author Waldemar Hummer
+ * @author Mark Fisher
+ */
+@Configuration
+@EnableBinding(Processor.class)
+@Import(SpelExpressionConverterConfiguration.class)
+@EnableConfigurationProperties(HttpClientProcessorProperties.class)
+public class HttpClientProcessorConfiguration {
+
+	@Bean
+	public RestTemplate restTemplate() {
```

Comment:
Well, then I'd like to see the documentation how to implement my own module.
And even better how to reuse an existing one without code copy/pasting.
As I said here: it would be great just leave everything as is but allow end-user to inject his own `RestTemplate`. Something like this:
````java
@SpringBootApplication(scanBasePackageClasses = HttpClientProcessorApplication.class)
public class MyHttpClientProcessorApplication {

	public static void main(String[] args) {
		SpringApplication.run(MyHttpClientProcessorApplication.class, args);
	}

       @Bean
	public RestTemplate restTemplate() { 
                 .........
        }
}
````